MESH-Research/CCR
# chore: add and configure mailhog for lando

* Add to Laravel config to detect MailHog and change from log to SMTP configuration for mail
* Add Lando configuration for running a MailHog container.
* Update the docs with info on the `lando extras` tooling command.

This pull request also adds a new Lando tooling command, `lando extras` to manage the local "ancillary" containers we're starting to accrue.  These containers are for things that we sometimes use, like majestic or cypress, but that including in the lando config and spinning up every time would make the developer experience poopy using up their machine's resources running containers that they may not need at the moment.  

Testing the MailHog setup:
1. Run `lando extras enable mailhog`
2. Run `lando rebuild`
3. Open the MailHog interface: http://mailhog.ccr.lndo.site/
4. Run `lando artisan tinker` and paste/run the following:
```
Mail::send('welcome', [], function($message) {
    $message->to('myawesomeemail@domain.com')->from("test@example.com")->subject('Testing mails');
});
```
5. Return to http://mailhog.ccr.lndo.site and confirm that the test email has arrived in the interface.

Cleaning up:
1. Run `lando extras disable mailhog` or `lando extras all off`
2. Run `lando rebuild` to remove the MailHog containers.
